### PR TITLE
Update samples for the In-Memory VectorStore Driver

### PR DESCRIPTION
**Pull request by joehu21**
Updated samples for the In-Memory VectorStore Driver, and added the missing PDF test file. Manually tested with MemoryDB 7.1.1 cluster.

### CODE REVIEW

File: samples/inmemory/retriever.ipynb [hunk 1]
```diff
@@ -10,7 +10,6 @@
     "import redis\n",
     "import os\n",
     "from dotenv import load_dotenv\n",
```
[the remark this answers is not in the document]
Comment (joehu21) [reply samples/inmemory/retriever.ipynb]: No longer needed. Removed.

File: samples/inmemory/retriever.ipynb [hunk 2]
```diff
@@ -40,9 +39,8 @@
     }
    ],
    "source": [
-    "\n",
     "from langchain_aws.embeddings import BedrockEmbeddings\n",
-    "from langchain_aws.llms.bedrock import Bedrock\n",
+    "from langchain_aws.llms.bedrock import BedrockLLM\n",
```
[the remark this answers is not in the document]
Comment (joehu21) [reply samples/inmemory/retriever.ipynb]: Tried changing it to ChatBedrock, but the RetrievalQA code below failed. I think the code below also needs to change. Will investigate when I get some free time. It is a learning opportunity for me. We can leave it as is for now.
```
qa_prompt = RetrievalQA.from_chain_type(
    llm=llm,
    chain_type="stuff",
    retriever=vds.as_retriever(),
    return_source_documents=True,
    chain_type_kwargs={"prompt": PROMPT},
    #verbose="true"
)
query = "How do i create a MemoryDB cluster"
result = qa_prompt({"query": query})
```

File: samples/inmemory/retriever.ipynb [hunk 3]
```diff
@@ -40,9 +39,8 @@
...
     "load_dotenv()"
```
[the remark this answers is not in the document]
Comment (joehu21) [reply samples/inmemory/retriever.ipynb]: Good to know.

File: samples/inmemory/retriever.ipynb [hunk 4]
```diff
@@ -118,11 +105,8 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "embeddings = BedrockEmbeddings(\n",
-    "        credentials_profile_name=os.environ.get(\"BWB_PROFILE_NAME\"), #sets the profile name to use for AWS credentials (if not the default)\n",
-    "        region_name=os.environ.get(\"BWB_REGION_NAME\"), #sets the region name (if not the default)\n",
-    "        endpoint_url=os.environ.get(\"BWB_ENDPOINT_URL\"), #sets the endpoint URL (if necessary)\n",
-    "    ) #create a Titan Embeddings client\n",
+    "# create a Titan Embeddings client\n",
+    "embeddings = BedrockEmbeddings()\n",
```
[the remark this answers is not in the document]
Comment (joehu21) [reply samples/inmemory/retriever.ipynb:97]: When model_id is not specified, the default one (amazon.titan-embed-text-v1) is used, which is good for this test.

File: samples/inmemory/retriever.ipynb [hunk 5]
```diff
@@ -83,24 +81,13 @@
...
+   "outputs": [],
    "source": [
-    "llm = Bedrock(\n",
-    "        credentials_profile_name=os.environ.get(\"BWB_PROFILE_NAME\"), #sets the profile name to use for AWS credentials (if not the default)\n",
-    "        region_name=os.environ.get(\"BWB_REGION_NAME\"), #sets the region name (if not the default)\n",
-    "        endpoint_url=os.environ.get(\"BWB_ENDPOINT_URL\"), #sets the endpoint URL (if necessary)\n",
-    "        model_id=\"anthropic.claude-v2\", #use the Anthropic Claude model\n",
+    "# use the Anthropic Claude model\n",
+    "llm = BedrockLLM(\n",
+    "        model_id=\"anthropic.claude-v2\",\n",
```
[the remark this answers is not in the document]
Comment (joehu21) [reply samples/inmemory/retriever.ipynb]: Tried to change it to claude-3, but it fails:
```
ValueError: Error raised by bedrock service: An error occurred (ValidationException) when calling the InvokeModel operation: The provided model identifier is invalid.
```
Here's my test:
```
from langchain_aws import BedrockLLM

# initialize the Bedrock LLM
llm = BedrockLLM(
    model_id="anthropic.claude-v3"
)

prompt = "What is the largest city in Vermont?"

# return a response to the prompt
response = llm.invoke(prompt)
print(response)
```

Any idea why the above fails?

BTW, langchain-aws [README](https://github.com/joehu21/langchain-aws) still uses claude-2.

File: samples/inmemory/retriever.ipynb [hunk 6]
```diff
@@ -225,13 +209,15 @@
...
      },
-     "execution_count": 9,
+     "execution_count": 11,
      "metadata": {},
      "output_type": "execute_result"
     }
    ],
    "source": [
-    "rc = MemoryDBCluster(host='Your_cluster_Endpoint', \n",
+    "endpoint = \"Your_Cluster_Endpoint\"\n",
```
[the remark this answers is not in the document]
Comment (joehu21) [reply samples/inmemory/retriever.ipynb]: like this. updated.

File: samples/inmemory/retriever.ipynb [hunk 7]
```diff
@@ -447,56 +432,59 @@
...
+      "Cluster naming constraints are as follows:\n",
+      "•Must contain 1–40 alphanumeric characters or hyphens.\n",
+      "•Must begin with a letter.\n",
+      "•Can't contain two consecutive hyphens.\n",
+      "•Can't end with a hyphen.\n",
+      "6. Under Encryption, choose whether to use a default encryption key or a customer managed key. For \n",
+      "more information, see In-transit encryption (TLS) in MemoryDB (p. 195).\n",
+      "7. Under Tags, optionally add tags to search and ﬁlter your snapshots or track your AWS costs.\n",
+      "130\n",
       "\n",
```
[the remark this answers is not in the document]
Comment (joehu21) [reply samples/inmemory/retriever.ipynb]: These are just output of the cell. Cleared outputs from all cells in the new commit.

File: samples/inmemory/retriever.ipynb (hunk 5 above)
[the remark this answers is not in the document]
Comment (3coins) [reply samples/inmemory/retriever.ipynb]: That model id seems wrong. You can use one of these. 
```
anthropic.claude-3-haiku-20240307-v1:0
anthropic.claude-3-opus-20240229-v1:0
anthropic.claude-3-sonnet-20240229-v1:0
```

Also, can you use ChatBedrock.

File: samples/inmemory/retriever.ipynb (hunk 2 above)
[the remark this answers is not in the document]
Comment (3coins) [reply samples/inmemory/retriever.ipynb]: RetrievalQA chain has been deprecated. Use this instead.

```python
from langchain.chains import create_retrieval_chain
from langchain.chains.combine_documents import create_stuff_documents_chain
from langchain_core.prompts import ChatPromptTemplate
from langchain_aws import ChatBedrock


retriever = ...  # Your retriever
llm = ChatBedrock(...)

system_prompt = (
    "Use the given context to answer the question. "
    "If you don't know the answer, say you don't know. "
    "Use three sentence maximum and keep the answer concise. "
    "Context: {context}"
)
prompt = ChatPromptTemplate.from_messages(
    [
        ("system", system_prompt),
        ("human", "{input}"),
    ]
)
question_answer_chain = create_stuff_documents_chain(llm, prompt)
chain = create_retrieval_chain(retriever, question_answer_chain)

chain.invoke({"input": query})
```

[the remark this answers is not in the document]
Comment (joehu21) [reply samples/inmemory/retriever.ipynb]: Still getting the same error:
```
ValueError: Error raised by bedrock service: An error occurred (ValidationException) when calling the InvokeModel operation: Malformed input request: #: subject must not be valid against schema {"required":["messages"]}#: extraneous key [max_tokens_to_sample] is not permitted, please reformat your input and try again.
```
code:
```
system_prompt = (
    "Use the given context to answer the question. "
    "If you don't know the answer, say you don't know. "
    "Use three sentence maximum and keep the answer concise. "
    "Context: {context}"
)
prompt = ChatPromptTemplate.from_messages(
    [
        ("system", system_prompt),
        ("human", "{input}"),
    ]
)
question_answer_chain = create_stuff_documents_chain(llm, prompt)
chain = create_retrieval_chain(retriever, question_answer_chain)
query = "How do i create a MemoryDB cluster?"
chain.invoke({"input": query})
```

[the remark this answers is not in the document]
Comment (joehu21) [reply samples/inmemory/retriever.ipynb]: Thanks for the offline help. Pushed the 3rd commit to change to ChatBedrock and ChatPromptTemplate.

File: samples/inmemory/retriever.ipynb (hunk 5 above)
[the remark this answers is not in the document]
Comment (joehu21) [reply samples/inmemory/retriever.ipynb]: updated

**Review by 3coins — APPROVED** [review]

**Pull request merged**